src/main.ts

```diff
@@ -22,11 +23,13 @@ export async function run() {
 
         let inputs = getInputs();
         let azureMySqlAction = new AzureMySqlAction(inputs);
-        let azureResourceAuthorizer = await AuthorizerFactory.getAuthorizer();
-        let azureMySqlResourceManager = await AzureMySqlResourceManager.getResourceManager(inputs.serverName, azureResourceAuthorizer);
-        firewallManager = new FirewallManager(azureMySqlResourceManager);
 
-        await firewallManager.addFirewallRule(inputs.serverName, inputs.connectionString);
+        if(!await MySqlUtils.connectsToDB(inputs.serverName, inputs.connectionString)) {
+            let azureResourceAuthorizer = await AuthorizerFactory.getAuthorizer();
+            let azureMySqlResourceManager = await AzureMySqlResourceManager.getResourceManager(inputs.serverName, azureResourceAuthorizer);
+            firewallManager = new FirewallManager(azureMySqlResourceManager);
```

Comment:
If the client cannot connect to MySql server, the `FirewallManager` will again do this check in `_detectIPAddress` method. Add logic to prevent re-checking.

Reply:
yes, thats right. in that case we need to refactor firewallmanager and make connectsToDB generic and use it.
Can we do this as a hotfix now and refactor later?

Reply:
removed duplicate code.